tests/nodejs_install.etest

```diff
@@ -0,0 +1,234 @@
+#!/usr/bin/env bash
+#
+# This program is free software: you can redistribute it and/or modify it under the terms of the Apache License
+# as published by the Apache Software Foundation, either version 2 of the License, or (at your option) any later
+# version.
+
+: ${EBASH_HOME:=$(dirname $0)/..}
+: ${EBASH:=${EBASH_HOME}/share}
+source "${EBASH}/ebash.sh" || { echo "Unable to source ${EBASH}/ebash.sh" ; exit 1 ; }
+
+# Import
+source "${EBASH_HOME}/install/nodejs"
+
+ETEST_clean_installation_check()
+{
+    emock "which" '
+    {
+        return 1
+    }'
+    check_for_node_installation
+}
+
+ETEST_cleanup_with_existing_symlinks()
+{
+    # Establish a test directory to install into
+    local test_prefix
+    test_prefix="$(mktemp -u usr_local.XXXXXX)"
+    local test_prefix_dir="${TEST_DIR_OUTPUT}/${test_prefix}"
+    local extracted_dir="${test_prefix_dir}/node-1.2.3.4"
+
+    # Simulate existing install
+    mkdir -p "${extracted_dir}/bin"
+    mkdir -p "${test_prefix_dir}/bin"
+    ln -sv "${extracted_dir}" "${test_prefix_dir}/node"
+
+    # Create binary symlinks
+    binaries=(
+        "node"
+        "npm"
+        "npx"
+        "nvm"
+    )
+
+    for binary in "${binaries[@]}"; do
+        echo "I think therefor I am" > "${extracted_dir}/bin/${binary}"
+        ln -sv "${test_prefix_dir}/node/bin/${binary}" "${test_prefix_dir}/bin/${binary}"
+    done
+
+    cleanup_node_symlinks --install_prefix "${test_prefix_dir}"
+
+    # All files should be cleaned up
+    for binary in "${binaries[@]}"; do
+        assert_not_exists "${test_prefix_dir}/bin/${binary}"
+    done
+
+    # Make sure top level node symlink is also removed
+    assert_not_exists "${test_prefix_dir}/node"
+}
+
+ETEST_cleanup_with_node_dir()
+{
+    # Establish a test directory to install into
+    local test_prefix
+    test_prefix="$(mktemp -u prefix_dir.XXXXXX)"
+    local test_prefix_dir="${TEST_DIR_OUTPUT}/${test_prefix}"
+    mkdir -p "${test_prefix_dir}/node"
+
+    try
+    {
+        cleanup_node_symlinks --install_prefix "${test_prefix_dir}"
```

Comment:
curious can your use assert_false here if you cleanup_node_symlinks to fail?

Reply:
Great suggestion, let me see about that!